### PR TITLE
Added spacing fix to catch end of line spaces

### LINKED ISSUE #47128: CatPhotoApp Step 6: Extra white space after closing tags causes step to fail
### Describe the Issue

The step that just keeps on giving. 

If user accidentally adds extra white space after `</h2>` or `-->` then step will not pass.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-6

### Your code

There is one space at the end of this line which causes step to fail.
```
<h2>Cat Photos</h2> 
```

There is one space at the end of this line which causes step to fail.
```
<!-- TODO: Add link to cat photos --> 
```

### Expected behavior

Step should pass even though there is extraneous white space at end of line.

### Screenshots

_No response_

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hanswang123456**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47128

Added fix for spacing after tags/comments in step 6 of cat-photo-app.

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dc23991f86c76b9248c6eb8.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Your comment should be below the `h2` element and start 6 spaces over from the s
 assert(
   code
     .toLowerCase()
-    .match(/<\/h2>\n\s{6}<!--\s*todo:\s+add\s+link\s+to\s+cat\s+photos\s*-->/)
+    .match(/<\/h2>\s+\n\s{6}<!--\s*todo:\s+add\s+link\s+to\s+cat\s+photos\s*-->/)
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dc23991f86c76b9248c6eb8.md:58]:
```suggestion
    .match(/<\/h2>\s*\n\s{6}<!--\s*todo:\s+add\s+link\s+to\s+cat\s+photos\s*-->/)
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dc23991f86c76b9248c6eb8.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ assert(
 Your `p` element should be below the comment and its opening tag should start 6 spaces over from the start of the line.
 
 ```js
-assert(code.toLowerCase().match(/-->\n\s{6}<p>/));
+assert(code.toLowerCase().match(/-->\s+\n\s{6}<p>/));
```
Comment (Sboonny) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dc23991f86c76b9248c6eb8.md:82]:
```suggestion
assert(code.toLowerCase().match(/-->\s*\n\s{6}<p>/));
```

**Review by Sboonny — CHANGES_REQUESTED** [review]
Test fails if someone doesn't add space after the text

image

**Comment by Sboonny** [comment]
Sorry for leaving you hanging, but I haven't approved the PR after the changes, because the issue is still in waiting triage and none else has toned and gave his opinion on it

**Comment by hanswang123456** [comment]
No worries, I had to redo a bit of the pull request and move it to a new branch on my fork.

**Pull request closed without merging**